website/server/libs/api-v3/payments.js

```diff
@@ -117,7 +118,8 @@ api.createSubscription = async function createSubscription (data) {
 api.cancelSubscription = async function cancelSubscription (data) {
   let plan = data.user.purchased.plan;
   let now = moment();
-  let remaining = data.nextBill ? moment(data.nextBill).diff(new Date(), 'days') : 30;
+  let defaultRemaining = plan.subscriptionLengthMonths ? plan.subscriptionLengthMonths * 30 : 30;
+  let remaining = data.nextBill ? moment(data.nextBill).diff(new Date(), 'days') : defaultRemaining;
```

Comment:
The reason I left nextBill in is because it still makes sense in some contexts. Payment methods that have subscription functionality look like they maintain this information for you so it makes sense that the subscription should end when they say you were supposed to pay next. Amazon doesn't have this feature and therefore shouldn't be sending a made up next date. 

The idea was to allow the payment method to override the default behavior if it had the information needed but to provide smart default behavior that should be capable of computing an accurate dateTerminated.

The main reason I would prefer not to remove this for all payment methods is that the default behavior depends on a new subscription being created with the new field. If we removed the nextBill from all payment methods, it would break anyone who tried to cancel an already existing subscription because the subscription wouldn't have a subscriptionLengthMonths field and would therefore default to 30 days even if paypal said that the next bill was a year from now.